sormas-ui/src/main/java/de/symeda/sormas/ui/person/PersonEditForm.java

```diff
@@ -631,6 +648,108 @@ protected void addFields() {
+		Date birthDate = calcBirthDateValue();
+		if (birthDate != null) {
+			Pair<Integer, ApproximateAgeType> pair =
+					ApproximateAgeHelper.getApproximateAge(birthDate, (Date) getFieldGroup().getField(PersonDto.DEATH_DATE).getValue());
+			if ((pair.getElement0() != null) && (pair.getElement1() == approximateAgeType)) {
+				return pair.getElement0();
+			}
+		}
+		return -1;
+	}
+
+	private void onEmancipatedChange() {
+		boolean isEmancipatedHidden = (isEmancipated == null) || (!isEmancipated.getValue());
+		boolean isEmancipatedChecked = (isEmancipated != null) && (isEmancipated.getValue());
+		hasGuardian.setValue(isEmancipatedHidden);
+		if(isEmancipatedHidden || isEmancipatedChecked) {
+			nameOfGuardians.setValue("");
+		}
+		updateHasGuardianCheckBox();
+	}
+
+	private void updateIsIncapacitatedCheckBox(boolean isInitialized) {
+		boolean isIncapacitatedChecked = (isIncapacitated == null) || (isIncapacitated.getValue());
+		isEmancipated.setVisible(!isIncapacitatedChecked && canBeEmancipated());
+		hasGuardian.setValue(isIncapacitatedChecked);
+		if(getApproximateAge(ApproximateAgeType.YEARS) < minimumAdultAge) {
+			nameOfGuardians.setValue(getValue().getNamesOfGuardians());
+		}
+		updateHasGuardianCheckBox();
+		hardResetNameOfGuardians(isInitialized);
+	}
+
+	private void hardResetNameOfGuardians(boolean isInitialized) {
+		boolean isIncapacitatedChecked = (isIncapacitated != null) && (isIncapacitated.getValue());
+		if (getApproximateAge(ApproximateAgeType.YEARS) != -1 && getApproximateAge(ApproximateAgeType.YEARS) >= minimumAdultAge && !isIncapacitatedChecked && isInitialized) {
+			nameOfGuardians.setValue("");
+		}
+	}
+
+	private void updateHasGuardianCheckBox() {
+		boolean isIncapacitatedChecked = (isIncapacitated != null) && (isIncapacitated.getValue());
+		boolean isEmancipatedChecked = (isEmancipated != null) && (isEmancipated.getValue());
+		if((getApproximateAge(ApproximateAgeType.YEARS) >= minimumAdultAge || (getApproximateAge(ApproximateAgeType.YEARS) < minimumEmancipatedAge) && isEmancipatedChecked)) {
+			isEmancipatedChecked = false;
+			isEmancipated.setValue(Boolean.FALSE);
+			isIncapacitated.setVisible(true);
+		}
+		if(isEmancipatedChecked){
+			hasGuardian.setVisible(false);
+			nameOfGuardians.setVisible(false);
+			isIncapacitated.setVisible(false);
+			return;
+		}
+		Date birthDate = calcBirthDateValue();
+		boolean isChildOrUnknownDate = isChildOrUnknownDate(birthDate);
+		nameOfGuardians.setVisible(isChildOrUnknownDate || isIncapacitatedChecked);
+		hasGuardian.setVisible(isChildOrUnknownDate || isIncapacitatedChecked);
+		if((birthDate == null) || isChildOrUnknownDate ){
+			boolean isEmancipatedVisible = ((isEmancipated != null) && (isEmancipated.isVisible()));
+			if(isEmancipatedVisible) {
+				hasGuardian.setValue(isChildOrUnknownDate);
+			} else {
+				hasGuardian.setValue(true);
+			}
+		}
+	}
+
+	private boolean isChildOrUnknownDate(Date birthDate) {
```

Comment:
this method is almost similar to getAproximateAge. Since it is used in only one place the logic can be included in "updateHasGuardianCheckBox" by calling getAproximateAge and add the condition regarding minimumAdultAge.